src/curl.c

```diff
@@ -58,12 +59,57 @@ uint64_t total_curl_sz = 0;
 /* alternative CA Path */
 static char *capath = NULL;
 
+static bool spinner_callback_enable = false;
+
 /* Pretty print curl return status */
 static void swupd_curl_strerror(CURLcode curl_ret)
 {
 	error("Curl - Download error - (%d) %s\n", curl_ret, curl_easy_strerror(curl_ret));
 }
 
+static char spinner[5] = { '|', '/', '-', '\\'};
+static time_t begin = 0;
+
+static int progress_spinner_callback(void UNUSED_PARAM *clientp, double UNUSED_PARAM dltotal,
+			     double UNUSED_PARAM dlnow, double UNUSED_PARAM ultotal, UNUSED_PARAM double ulnow)
+{
+#define MAX_THRESHOLD_MULTIPLE_SIX 36000
+
+	if (begin ==0){
+		begin = time(NULL);
+	}
+	
+	if ((time(NULL) - begin) > 0.5) {
+		begin = time(NULL);
+		static int index;
```

Comment:
I would move this declaration to the top of the function for clarity as it's a static variable.

Reply:
Done